### ISSUE TITLE
[KED-1172] Enabling different types of encryption for s3 dataset types. 

### ISSUE THREAD

**Issue by williamashfordQB** [quantumblacklabs/kedro#181]
## Description
As an engineer I would like to be able to change the means of encryption on s3 datasets in order to save files on s3 in accordance with client information security requirements.  

## Context
When saving files to s3, some organisation require certain types of server-side encryption, namely amazon's Key Management System (KMS). As it is not easy to enable this in the s3 datasets, users are need to subclass the datasets. 

## Possible Implementation
Allow users to access the S3FileSystem on class instantiation, specifically, something to the tune of:
```
def init(credentials, s3fs_args, ...):
    _s3 = S3FileSystem(client_kwargs=_credentials, **s3fs_args)
   ...
```

## Possible Alternatives
A singleton pattern to allow users to access the s3FileSystem object and set defaults prior to instantiation of the catalog.

**Comment by lorenabalan** [comment]
As discussed on Slack, this is already work in progress. Will update this issue with the relevant commit when it's merged.

**Comment by williamashfordQB** [comment]
Yep no worries. Just thought I'd log here to follow the process.

**Comment by ZainPatelQB** [comment]
Thanks for logging this @williamashfordQB, it's been addressed in https://github.com/quantumblacklabs/kedro/commit/f4f66f70beab6c04e087e4aaf6acec69d8181d34 and available in `0.15.5` 🎉

I'm going to go ahead and close this issue. :)
